Adding rebate rate and fee rate.

ashutoshrana171 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Description
We added rebate rate and fee rate.

We added rebate rate and fee rate provided by InteractiveBrokersShortableProvider by extending the IBKR Shortable reader to parse the rebate and fee rate

#### Related Issue
Close [7416](https://github.com/QuantConnect/Lean/issues/7614)
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

#### Motivation and Context
 We added rebate and fee rate supported by Interactive Brokers

#### Requires Documentation Change
No

#### How Has This Been Tested?
We used a sample file from InteractiveBrokersShortableProvider and processed it for testing.
Sample file:
[usa.txt](https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/files/14202487/usa.txt)
New File:
[20240207.csv](https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/files/14202492/20240207.csv)

#### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
<!--- The following is a checklist of items that MUST be completed before a PR is accepted -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] I have added tests to cover my changes. <!--- If not applicable, please explain why -->
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

<!--- Template inspired by https://www.talater.com/open-source-templates/#/page/99 -->

ashutoshrana171 committed:
Adding rebate rate and fee rate.

AlexCatarino reviewed (approved):
Thank you, @ashutoshrana171 !

Martin-Molinero commented on DataProcessing/DataConverter.cs:
```diff
@@ -130,17 +132,31 @@ private void Process(DateTime date, string line)
 
             var ticker = csv[_symbolIndex].Replace(' ', '.');
             var borrowableShares = csv[_availableIndex];
+            var rebateRateIndex = csv[_rebateRateIndex];
+            var feeRateIndex = csv[_feeRateIndex];
```
We shouldn't assume `_rebateRateIndex/_feeRateIndex` will be there always, can happen for older dates, so here we should check if they are different than -1 before trying to access them

Martin-Molinero commented on DataProcessing/DataConverter.cs:
```diff
@@ -282,8 +298,12 @@ public void TryAdd(DateTime date, string borrowableShares)
             {
-                return _entries.OrderBy(kvp => kvp.Key)
-                    .Select(kvp => string.Join(",", kvp.Key.ToString("yyyyMMdd"), kvp.Value));
+                return _entries.OrderBy(entry => entry.Item1)
```
I think now that we have more columns, in the output file we should add a header with the columns names too, for both date and symbol files 👍. Lean will already handle (skip) comment lines starting with `#`  see https://github.com/QuantConnect/Lean/blob/master/Common/Data/Shortable/LocalDiskShortableProvider.cs#L101

ashutoshrana171 committed:
- Adding as per peer
   - Added check for rebate and fee for -1
   - Added headers for data and symbol files

Martin-Molinero commented:
Replaced by https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/pull/3
